packages/runtime-common/realm.ts

```diff
@@ -228,21 +236,84 @@ export class Realm {
     await this.ready;
   }
 
+  async flushUpdateEvents() {
+    return this.#flushUpdateEvents;
+  }
+
   async write(
     path: LocalPath,
     contents: string,
   ): Promise<{ lastModified: number }> {
+    await this.trackOwnWrite(path);
     let results = await this.#adapter.write(path, contents);
-    await this.#searchIndex.update(this.paths.fileURL(path));
+    await this.#searchIndex.update(this.paths.fileURL(path), {
+      onInvalidation: (invalidatedURLs: URL[]) => {
+        this.sendUpdateMessages({
+          type: 'update',
+          data: { 'index-invalidation': JSON.stringify(invalidatedURLs) },
```

Comment:
Would it make sense to put "index-invalidation" value in an enum of event names?